Tiltfile

```diff
@@ -0,0 +1,93 @@
+load("ext://restart_process", "docker_build_with_restart")
+load("ext://helm_resource", "helm_resource", "helm_repo")
+load("ext://configmap", "configmap_create")
+
+# A list of directories where changes trigger a hot-reload of PATH.
+# Note: this list needs to be updated each time a new package is added to the repo.
+hot_reload_dirs = ["cmd", "config", "gateway", "health", "message", "qos", "relayer", "request", "router"]
+
+# Load the existing config file, if it exists, or use an empty dict as fallback
+localnet_config_path = "localnet_config.yaml"
+localnet_config = read_yaml(localnet_config_path, default={})
+
+# TODO_UPNEXT(@adshmh): Package the default Helm chart for PATH and upload the pokt-network repo.
+# Configure helm chart reference. If using a local repo, set the path to the local repo; otherwise, use our own helm repo.
+helm_repo("pokt-network", "https://pokt-network.github.io/helm-charts/")
+chart_prefix = "pokt-network/"
+if localnet_config["helm_chart_local_repo"]["enabled"]:
+    helm_chart_local_repo = localnet_config["helm_chart_local_repo"]["path"]
+    hot_reload_dirs.append(helm_chart_local_repo)
+    print("Using local helm chart repo " + helm_chart_local_repo)
+    chart_prefix = helm_chart_local_repo + "/charts/"
+
+# TODO_TECHDEBT(@adshmh): use secrets for sensitive data with the following steps:
+# 1. Add place-holder files for sensitive data
+# 2. Add a secret per sensitive data item (e.g. gateway's private key)
+# 3. Load the secrets into environment variables of an init container
+# 4. Use an init container to run the scripts for updating config from environment variables.
+# This can leverage the scripts under `e2e` package to be consistent with the CI workflow.
+
+# Import configuration files into Kubernetes ConfigMap
+configmap_create("path-config", from_file="localnet/path/config/.config.yaml", watch=True)
+
+# Build an image with a poktrolld binary
+docker_build_with_restart(
+    "path",
+    ".",
+    dockerfile="Dockerfile",
+    entrypoint="/app/path",
+    live_update=[sync("bin/path", "/app/path")],
+)
+
+# Provision PATH
+helm_resource(
+    "path",
+    chart_prefix + "path",
+    # TODO_UPNEXT(@adshmh): Add the CLI flag for loading the configuration file, once the CLI flags feature has been implemented.
```

Comment:
OPTIONAL NIT: Consider using `TODO_MVP` if it's not actually "Up next".

This is something new I learnt as we're working towards Beta TestNet.

Feel free to ignore though. I'm iterating on this approach alongside everyone else.